x/rollapp/keeper/msg_server_update_state.go

```diff
@@ -65,6 +71,11 @@ func (k msgServer) UpdateState(goCtx context.Context, msg *types.MsgUpdateState)
 	// Write new state information to the store indexed by <RollappId,LatestStateInfoIndex>
 	k.SetStateInfo(ctx, *stateInfo)
 
+	err = k.hooks.AfterUpdateState(ctx, msg.RollappId, stateInfo, isFirstStateUpdate, previousStateHasTimestamp)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I think the 'toggle' to switch on timestamps should be enforced in the rollapp keeper (this file) rather than the light client keeper
The timestamp is used by the light client feature but it doesn't need to be coupled to it